app/src/main/assets/extensions/wolvic_autowebxr/content.js

```diff
@@ -0,0 +1,142 @@
+const LOGTAG = '[wolvic:autowebxr]';
+const ENABLE_LOGS = true;
+const logDebug = (...args) => ENABLE_LOGS && console.log(LOGTAG, ...args);
+
+const PARENT_ELEMENT_XPATH_PARAMETER = 'wolvic-autowebxr-parentElementXPath';
+const TARGET_ELEMENT_XPATH_PARAMETER = 'wolvic-autowebxr-targetElementXPath';
+
+const IFRAME_READY_MSG = 'wolvic-autowebxr-iframeReady';
+const TARGET_ELEMENT_MSG = 'wolvic-autowebxr-targetElement';
+
+var parentElementXPath;
+var targetElementXPath;
+
+function getElementByXPath(document, xpath) {
+    let result = document.evaluate(xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null);
+    return result.singleNodeValue;
+}
+
+function clickImmersiveElement() {
+    // Check if the current URL has extra query parameters
+    parentElementXPath = undefined;
+    targetElementXPath = undefined;
+
+    let url = document.URL;
+    let params = new URLSearchParams(new URL(url).search);
+    for (let [key, value] of params) {
+        if (key === PARENT_ELEMENT_XPATH_PARAMETER)
+            parentElementXPath = value;
+        else if (key === TARGET_ELEMENT_XPATH_PARAMETER)
+            targetElementXPath = value;
+    }
+
+    // We need at least the target element to click
+    if (!targetElementXPath)
+        return;
+
+    logDebug('Preparing to open immersive WebXR; parentElementXPath: ' + parentElementXPath + ' ; targetElementXPath: ' + targetElementXPath);
+
+    // The parent element is typically an iframe and, if it comes from a different origin,
+    // we might not be able to access its contents directly.
+    // If parentElementXPath is null, we will use the root to find the target element.
+
+    var parent, parentDocument;
+    if (parentElementXPath) {
+        parent = getElementByXPath(document, parentElementXPath);
+        if (!parent) {
+            logDebug('Parent element not found, retrying');
+            setTimeout(clickImmersiveElement, 1000);
```

Comment:
This has the potential of trigger a never ending sequence of timeouts if the parent element is not correctly specified. I guess you added the timeout to allow the page to load, but in that case I think it's safer to listen to DOMContentLoaded or similar event.

Reply:
Note that this code is called after document.readyState === 'complete' or the load event has been already received. The parent might still be unreachable because the website is still setting things up via JS.

My idea here was simply to call again clickImmersiveElement after a small delay. Maybe it makes sense to have a timer or a counter so we don't keep trying forever.

Reply:
Ah right. In any case I think we need a counter there to prevent the potential never ending loop.